psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -85,6 +85,70 @@ Wavefunction::Wavefunction(std::shared_ptr<Molecule> molecule, std::shared_ptr<B
     common_init();
 }
 
+Wavefunction::Wavefunction(std::shared_ptr<Molecule> molecule, std::shared_ptr<BasisSet> basisset, 
+                           std::map<std::string, std::shared_ptr<Matrix>> matrices,
+                           std::map<std::string, std::shared_ptr<Vector>> vectors,
+                           std::map<std::string, Dimension> dimensions, std::map<std::string, int> ints, 
+                           std::map<std::string, std::string> strings, std::map<std::string, bool> booleans, 
+                           std::map<std::string, float> floats)
+    : options_(Process::environment.options),
+      basisset_(basisset),
```

Comment:
Does this build the sobasisset as well?

Reply:
Not yet, about to add. Do you think it would be better to try and save the SO basis or recreate it in the same way that the normal constructor does?

https://github.com/psi4/psi4/blob/3c26e8eb81c20bce3fe0223d44606c5f9be59b81/psi4/src/psi4/libmints/wavefunction.cc#L376-L379

Reply:
The above two quantities can be deterministically created from a basisset, no need to save.